### PR TITLE
Direct deposit definition

### LINKED ISSUE #15843: Development for direct deposit component
## Issue Description
Development for direct deposit component 

---
## Tasks
- [ ] Take artifact from design and implement 
- [ ] Unit tests
- [ ] Stories/description of what the component does 

## Acceptance Criteria
- [ ] Sign off from designer
---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by cvalarida**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/15843

## Testing done
- Wrote unit tests
- Created a new form and used it there to verify visually

## Screenshots

<details>
<summary>All the fields</summary>

image
</details>

<details>
<summary>Without the optional fields</summary>

image
</details>

<details>
<summary>Direct deposit declined</summary>

image
</details>

<details>
<summary>Additional info content</summary>

image
</details>

<details>
<summary>Pre-filled bank information</summary>

image
</details>

<details>
<summary>Edited bank information</summary>

image
</details>

## Acceptance criteria
- [x] Bank name is optional
- [x] Decline direct deposit is optional
- [x] All the fields appear in the correct order
- [x] Declining the direct deposit hides the rest of the fields

### CODE REVIEW

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import AdditionalInfo from '@department-of-veterans-affairs/formation-react/AdditionalInfo';
+
+export const directDepositDescription = () => {
+  return (
+    <div>
+      <p>
+        We make payments only through direct deposit, also called electronic
+        funds transfer (EFT). If you’re approved for the Rogers STEM
+        Scholarship, we’ll need to know where to deposit the funds.
```
Comment (CrystabelReiter) [inline src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:10]: Can we remove the sentence "If you're approved for the Rogers STEM Scholarship, we'll need to know where to deposit the funds." @cvalarida

**Review by bkjohnson — COMMENTED** [review]
I need to understand fields/widgets a bit more :upside_down_face:

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx (hunk 1 above)
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:10]: :joy: Yeah, that...yeah. :see_no_evil: :sweat_smile:

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      information for VA benefits, including disability compensation, pension,
+      and education. These updates won’t change your information for VA health
+      care.
+    </p>
+  </div>
+);
+
+export const bankInfoHelpText = (
+  <AdditionalInfo
+    triggerText="What if I don’t have a bank account or don’t want to use direct deposit?"
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:41]: I don't think so. The "extra" info attached to the GA event is going to be different, but I think that's fine.

Now that you mention it, though, I need to remove that `edu-10203--` :sweat_smile:

File: src/platform/forms-system/src/js/definitions/directDeposit.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+ * @param {bool|OptionalField} [optionalFields.declineDirectDeposit] - Set to true
+ *        to add the declineDirectDeposit field the default schema and uiSchema.
+ *        Pass an object with schema and uiSchema properties to override either
+ *        or both.
+ * @param {bool|OptionalField} [optionalFields.bankName] - Set to true to add the
+ *        bankName field with the default schema and uiSchema. Pass an object
+ *        with schema and uiSchema properties to override either or both.
+ */
+export default optionalFields => {
+  const opts = merge({}, defaultOptionalFields, optionalFields);
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js:158]: Good call. I think this is an artifact from a different approach. I was going to make the argument passed into the function more than just optional fields, but turns out I didn't need to.

Removing!

File: src/platform/forms-system/src/js/definitions/directDeposit.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+          accountType: {
+            type: 'string',
+            enum: ['checking', 'savings'],
+          },
+          routingNumber: {
+            type: 'string',
+            pattern: '^\\d{9}$',
+          },
+          accountNumber: {
+            type: 'string',
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js:112]: Maybe? I honestly don't know why it is this way. I just copied it from another schema. :see_no_evil: 

I'm not against making it a number, but I don't know what the ramifications of that would be, so I'm not especially comfortable with it in this PR.

File: src/platform/forms-system/src/js/definitions/directDeposit.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        'ui:required': useDirectDeposit,
+      },
+      accountNumber: {
+        ...bankAccountUI.accountNumber,
+        'ui:title': 'Bank account number',
+        'ui:required': useDirectDeposit,
+      },
+      'ui:options': {
+        classNames: 'vads-u-margin-bottom--3',
+        hideIf: form => form?.declineDirectDeposit,
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js]: 100% could! Good call.

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+const gaBankInfoHelpText = () => {
+  window.dataLayer.push({
+    event: 'form-help-text-clicked',
+    'help-text-label': 'What if I don’t have a bank account?',
+  });
+};
+
+export const directDepositAlert = () => (
+  // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:30]: Hmm...I just copied this from the 10203. @1Copenut thoughts?

File: src/platform/forms-system/src/js/definitions/directDeposit.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        'ui:required': useDirectDeposit,
+      },
+      accountNumber: {
+        ...bankAccountUI.accountNumber,
+        'ui:title': 'Bank account number',
+        'ui:required': useDirectDeposit,
+      },
+      'ui:options': {
+        classNames: 'vads-u-margin-bottom--3',
+        // eslint-disable-next-line react-hooks/rules-of-hooks
```
[the remark this answers is not in the document]
Comment (cvalarida) [reply src/platform/forms-system/src/js/definitions/directDeposit.js:51]: It's not an actual hook. Just a function that starts with `use`. :man_shrugging:

**Review by bkjohnson — APPROVED** [review]
Seems fine to me :+1:

**Comment by cvalarida** [comment]
**Status update:** Waiting on content review from @DanielleThierryUSDSVA

File: src/platform/forms-system/src/js/definitions/content/directDeposit.jsx [hunk 8]
```diff
@@ -26,14 +26,17 @@ const gaBankInfoHelpText = () => {
...
   // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
   <div className="vads-u-padding-top--1p5" tabIndex="0">
     <p>
       <b>Note:</b> Any updates you make here will change your bank account
-      information for VA benefits, including disability compensation, pension,
-      and education. These updates won’t change your information for VA health
-      care.
+      information for VA benefits, including{' '}
+      {affectedBenefits || '[affected benefits]'}. These updates won’t change
+      your information for {unaffectedBenefits || '[unaffected benefits]'}.
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/definitions/content/directDeposit.jsx:39]: Are the brackets inside the string part of the mockup? Is this something that we want/expect veterans to see?

**Comment by cvalarida** [comment]
Updated content:
image

The link opens https://veteransbenefitsbanking.org/ in a new tab.

**Review by CrystabelReiter — APPROVED** [review]

File: src/platform/forms-system/test/js/components/PaymentView.unit.spec.jsx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+    view.getByText('We’re currently paying your compensation to this account');
+    view.getByText('Checking Account');
+    expect(view.getByTestId('account-number').textContent).to.equal(
+      'Account number: ●●●●●●ending with4321',
+    );
+    expect(view.getByTestId('routing-number').textContent).to.equal(
+      'Bank routing number: ●●●●●ending with6789',
+    );
+    expect(view.getByTestId('bank-name').textContent).to.equal(
+      'Bank name: Gringotts',
```
Comment (bkjohnson) [inline src/platform/forms-system/test/js/components/PaymentView.unit.spec.jsx:30]: Yer a wizard!

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**